src/routes/friend.js

```diff
@@ -1,88 +1,164 @@
-      res.json(student.friends).status(200)
-    })
-    .catch(err => res.status(400).json('Error: ' + err));
+      .then((student) => {
+        res.json(student.friends).status(200)
+      })
+      .catch(err => res.status(400).json('Error: ' + err));
 })
 
-// send  a friend rerquest
-router.route('/add').post(async (req, res) => {
-  const username = req.body.username.toString();
-  const friendUsername = req.body.friendUsername.toString()
-  StudentRepository.findOneByUsername(username)
-    .then((myProfile) => {
-      StudentRepository.findOneByUsername(friendUsername).then((friendProfile) => {
-        let friend = friendUsername;
-        let me = username;
-
-        let friendRequestsSent = myProfile.friendRequestsSent
-        friendRequestsSent.push(friendUsername)
-
-        let friendRequestsReceived = friendProfile.friendRequestsReceived
-        friendRequestsReceived.push(username)
-
-        StudentRepository.updateFriendRequestSent(me, friendRequestsSent)
-          .then()
-          .catch(err => res.status(400).json('Error: ' + err));
-
-        StudentRepository.updateFriendRequestReceived(friend, friendRequestsReceived)
-          .then()
-          .catch(err => res.status(400).json('Error: ' + err));
-      })
-        .catch(err => res.status(400).json('Error: ' + err));
-    })
-    .catch(err => res.status(400).json('Error: ' + err));
-});
+/**
+ * @author: Jasmin Guay
+ * Endpoint to update a student friend list
+ * @param {string} email : email of the student to be updated
+ * @param {string} friends : list of updated friend list.
+ * @return {[string]} the update friend list.
+ */
+router.route('/updateFriendList').post( async (req,res) => {
+    const email = req.body.email.toString();
+    const updatedFriendList = req.body.friends.toString();
+
+    StudentRepository.updateFriendList(email,updatedFriendList)
+        .then((student) => { res.json(student.friends).status(200) })
+        .catch((err) => { res.json(`Error happened in updateFriendList - ${err}`).status(404) })
+})
 
-router.route('/answer').post(async (req, res) => {
 
-  const username = req.body.username.toString();
-  const friendUsername = req.body.friendUsername
-  const answer = req.body.answer;
+/**
+ * @author: Jasmin Guay
+ * Endpoint to answer a friend request.
+ * @param {string} requestId  id of the request.
+ * @param {"accepted" | "declined"} answer : answer to the friend request.
+ * @return {string || null} studentUsername : username of the student added to the friend list if the request was accepted, null otherwise.
+ */
+router.route('/answerFriendRequest').post(async (req, res) => {
+  const requestId = req.body.requestId.toString();
+  const user = req.body.user;
+  const request = await FriendRequest.findOne({_id: requestId});
 
-  Student.updateOne({ username: username },
-    { $pull: { friendRequestsReceived: { $in: [friendUsername] } } })
+  if (!validateFriendRequest(request, user) || request == null) {
```

Comment:
You should probably reverse the order of this condition (i.e. if(request === null || !validateFriendRequest(request,user) {...}). This is to ensure that the validateFriendRequest method is not called when the request is null (to avoid runtime errors).